src/v4l2_camera_device.cpp

```diff
@@ -296,6 +287,16 @@ Image::UniquePtr V4l2CameraDevice::capture()
 
   auto const & buffer = buffers_[buf.index];
   std::copy(buffer.start, buffer.start + img->data.size(), img->data.begin());
+
+  // Requeue buffer to be reused for new captures
+  if (-1 == ioctl(fd_, VIDIOC_QBUF, &buf)) {
+    RCLCPP_ERROR(
+      rclcpp::get_logger("v4l2_camera"),
+      "Error re-queueing buffer: %s (%s)", strerror(errno),
+      std::to_string(errno).c_str());
+    return nullptr;
+  }
+
```

Comment:
@pfpacket 
Although the modification looks good, I would like to ask you for further improvement. Could you consider to move some field assignments after re-queue so that re-queue happens as soon as possible?
```suggestion
  img->header.stamp = buf_stamp;
  img->width = cur_data_format_.width;
  img->height = cur_data_format_.height;
  img->step = cur_data_format_.bytesPerLine;
  if (cur_data_format_.pixelFormat == V4L2_PIX_FMT_YUYV) {
    img->encoding = sensor_msgs::image_encodings::YUV422_YUY2;
  } else if (cur_data_format_.pixelFormat == V4L2_PIX_FMT_UYVY) {
    img->encoding = sensor_msgs::image_encodings::YUV422;
  } else if (cur_data_format_.pixelFormat == V4L2_PIX_FMT_GREY) {
    img->encoding = sensor_msgs::image_encodings::MONO8;
  } else {
    RCLCPP_WARN(rclcpp::get_logger("v4l2_camera"), "Current pixel format is not supported yet");
  }
```

Reply:
@manato Pushed the fix. Now the commit is based on the one from the upstream with conflicts resolved.